src/index.d.ts

```diff
@@ -0,0 +1,12 @@
+import { IncomingRequest } from 'http'
+
+type Options = {
+  limit: number
+}
+
+export default {
+  buffer(req: IncomingRequest, options: Options): typeof buffer;
+  string(req: IncomingRequest, options: Options): string;
+  json(req: IncomingRequest, options: Options): object;
+  urlencoded(req: IncomingRequest, options: Options): object;
```

Comment:
```suggestion
  buffer(req: IncomingRequest, options?: Partial<Options>): typeof buffer;
  string(req: IncomingRequest, options?: Partial<Options>): string;
  json(req: IncomingRequest, options?: Partial<Options>): object;
  urlencoded(req: IncomingRequest, options?: Partial<Options>): object;
```

but what if instead you used JSDoc? TypeScript can parse these, which is what Svelte is doing.

Reply:
also, can `urlencoded` return `URLSearchParams`?

Reply:
can `json` return `unknown`?

Reply:
can `buffer` return ArrayBuffer? or `Uint8Array`? or `Buffer`?

Reply:
<picture data-single-emoji=":lol:" title=":lol:"><img class="emoji" width="20" height="auto" src="https://single-emoji.vercel.app/api/emoji/eyJhbGciOiJkaXIiLCJlbmMiOiJBMjU2R0NNIn0..mrGGuL-m2xd3h_ed.zWiyOpK6BZucQgow8qhTharQJIwS0uLmuIlcKQpcQXbMcmjpdY93LAuoivB2LxYkmJwqgbXnXQn39biwyfxGujySkjNFnYASCKE6Chy3VzsOLkI.C8Yz7Ij0r2uHDuZbx7HfzA" alt=":lol:" align="absmiddle"></picture>